src/components/common/YourCompany/AssociationType.vue

```diff
@@ -0,0 +1,256 @@
+                    <v-list-item-subtitle>
+                      <v-icon small color="primary">mdi-pencil</v-icon>
+                      <span class="drop-down-action ml-1">Change</span>
+                    </v-list-item-subtitle>
+                  </v-list-item>
+                </v-list>
+              </v-menu>
+            </span>
+        </div>
+      </v-col>
+
+      <!-- Editing Mode -->
+      <v-col cols="9" v-if="isEditingAssociationType">
+        <v-select
+          filled
+          id="association-type-input"
+          :items="associationTypeOptions"
+          v-model="selectedAssociationType"
+          :rules="AssociationTypeRules"
+        >
+        </v-select>
+
+        <!-- Done Actions -->
+        <div class="action-btns">
+          <v-btn
+            id="done-btn"
+            large color="primary"
+            @click="submitAssociationTypeChange()"
+          >
+            <span>Done</span>
+          </v-btn>
+          <v-btn
+            id="cancel-btn"
+            large outlined color="primary"
+            @click="isEditingAssociationType = false"
+          >
+            <span>Cancel</span>
+          </v-btn>
+        </div>
+      </v-col>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins, Prop, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'vuex-class'
+import { AssociationTypes } from '@/enums'
+import { CommonMixin, EnumMixin } from '@/mixins'
+import { ActionBindingIF, BusinessInformationIF, EntitySnapshotIF } from '@/interfaces'
+
+@Component({})
+export default class AssociationType extends Mixins(CommonMixin, EnumMixin) {
+  @Prop({ default: false })
+  readonly invalidSection: boolean
+  /** Global getters */
+  @Getter getAssociationType!: AssociationTypes
+  @Getter getBusinessInformation!: BusinessInformationIF
+  @Getter getEntitySnapshot!: EntitySnapshotIF
+  @Getter hasAssociationTypeChanged!: boolean
+
+  /** Global actions */
+  @Action setBusinessInformation!: ActionBindingIF
+
+  /** Select options */
+  readonly associationTypeOptions: Array<any> = [
+    {
+      value: AssociationTypes.COMMUNITY_SERVICE_COOPERATIVE,
+      text: this.associationTypeToDescription(AssociationTypes.COMMUNITY_SERVICE_COOPERATIVE)
+    },
+    {
+      value: AssociationTypes.ORDINARY_COOPERATIVE,
+      text: this.associationTypeToDescription(AssociationTypes.ORDINARY_COOPERATIVE)
+    },
+    {
+      value: AssociationTypes.HOUSING_COOPERATIVE,
+      text: this.associationTypeToDescription(AssociationTypes.HOUSING_COOPERATIVE)
+    }
+  ]
+
+  protected selectedAssociationType = null as AssociationTypes
+  protected isEditingAssociationType = false
+
+  /** V-model for dropdown menu. */
+  protected dropdown: boolean = null
+
+  /** Validation rules. */
+  readonly AssociationTypeRules: Array<Function> = [
+    v => !!v || 'This field is required' // is not empty
+  ]
+
+  mounted () {
+    this.initializeAssociationType()
```

Comment:
No change needed but FYI: you could also init the association type by adding `{ immediate: true}` as a watch parameter. This will cause the watcher to fire when the component is created/mounted.

I'm OK with this code. It's clear.